src/actions/variable.ts

```diff
@@ -34,14 +35,69 @@ import {ExpressionQualifier} from "../qualifiers/expression/ExpressionQualifier"
  * @summary action
  * @description Sets a new user variable with the given value.
  * @memberOf Actions.Variable
- * @param name Variable name
+ * @param {string} name Variable name
  * @param {number | string | number[] | string[]} value Variable value
  * @return {Actions.Variable.SetAction}
  */
 function set(name: string, value: number | string | number[] | string[] | ExpressionQualifier): SetAction {
+  if (Object.prototype.hasOwnProperty.call(value, 'push')) {
+    return new SetAction(name, value);
+  }
+
   return new SetAction(name, value);
 }
 
+/**
+ * @summary action
+ * @description Same as 'set', but forces the end value to be a float  setFloat(1) will result in $foo_1.0
+ * @memberOf Actions.Variable
+ * @param {string} name Variable name
+ * @param {number} value Variable value
+ * @return {Actions.Variable.SetAction}
+ */
+function setFloat(name: string, value: number): SetAction {
+  return new SetAction(name, toFloatAsString(value), '');
+}
+
+/**
+ * @summary action
+ * @description Same as 'set', but forces the end value to be an integer setInteger(1.1) will result in $foo_1, input is rounded down
+ * @memberOf Actions.Variable
+ * @param {string} name Variable name
+ * @param {number} value Variable value
+ * @return {Actions.Variable.SetAction}
+ */
+function setInteger(name: string, value: number): SetAction {
+  let val = value;
+  if (typeof value === 'string') {
+    val = parseInt(value);
+  }
+
+  if (isNaN(val)) {
+    val = 0;
+  }
+
+  return new SetAction(name, Math.round(val));
+}
+
+
+/**
+ * @summary action
+ * @description Same as 'set', but forces the end value to be a string setString(1) will result in $foo_!1!
+ * @memberOf Actions.Variable
+ * @param {string, number} name Variable name
+ * @param {number} value Variable value
+ * @return {Actions.Variable.SetAction}
+ */
+function setString(name: string, value: string | number): SetAction {
```

Comment:
Why allow number?

Reply:
I thought it might be a good idea to support it as well, but we can make it strictly a string.

In that case I wonder if we really need the `setString()` function, as `set('foo')` and `setString('foo')` do the same thing and are not ambigious.

What do you think?

Reply:
So lets leave it like you did